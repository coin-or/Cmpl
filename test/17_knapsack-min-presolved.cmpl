
	include "knapsack-data.cmpl"
	
	#calculating the relative value of each box	
	{j in boxes: val[j]:= w[j]/p[j]; } 
	
	M:=10000;
	sumSales:=0;
	sumWeight:=0;
	#initial solution 
	x[]:=(0,0,0,0,0,0,0,0,0,0);
	{sumSales < minSales: 
		maxVal:=min(val[]);
		{j in boxes: 
			{ maxVal=val[j] : 
					{ sumSales < minSales: 
							x[j]:=1;
							sumSales:=sumSales + p[j];
							sumWeight:=sumWeight + w[j]; 
					}
					val[j]:=M;
					break j;
			}
		}	
		repeat;
	}  
	echo "Solution found";
	echo "Optimal total weight : " + sumWeight;
	echo "Total sales: "+ sumSales;	
	{j in boxes: echo "x_"+ j + ": " + x[j]; }
