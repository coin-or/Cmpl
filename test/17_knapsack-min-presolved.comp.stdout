Solution found
Optimal total weight : 47
Total sales: 630
x_1: 1
x_2: 1
x_3: 0
x_4: 1
x_5: 0
x_6: 0
x_7: 0
x_8: 0
x_9: 1
x_10: 1
